### PR TITLE
[HINTS] Add implementation for hints on blake2s_add_uint256 & blake2s_add_uint256_bigend (cairo_blake2s)

### PR DESCRIPTION
**Pull request by fmoletta**
- Add implementation for hints on `blake2s_add_uint256` and `blake2s_add_uint256_bigend`
- Add integration test using `blake2s_felts`
-Add unit tests for both hints

### CODE REVIEW

File: cairo_programs/blake2s_felts.cairo [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    assert inputs[6] = 87158958
+    assert inputs[7] = 6353668
+    assert inputs[8] = 343656565
+    assert inputs[9] = 1255962
+    assert inputs[10] = 25439785
+    assert inputs[11] = 1154578
+    assert inputs[12] = 585849303
+    assert inputs[13] = 763502
+    assert inputs[14] = 43753647
+    assert inputs[15] = 74256930
```
Comment (Oppen) [inline cairo_programs/blake2s_felts.cairo:26]: Is this an arbitrary input or does it represent something in particular?

File: src/vm/hints/blake2s_utils.rs [hunk 2]
```diff
@@ -144,6 +152,110 @@ pub fn finalize_blake2s(
...
+    let low = &vm.memory.get_integer(&low_addr)?.clone();
+    let high = &vm.memory.get_integer(&high_addr)?.clone();
+    //Main logic
+    //Declare constant
+    const MASK: u32 = u32::MAX;
+    const B: u32 = 32;
+    //Build first batch of data
+    let mut inner_data = Vec::<BigInt>::new();
+    for i in 0..4 {
+        inner_data.push((low >> (B * i)) & bigint_u32!(MASK));
```
Comment (Oppen) [inline src/vm/hints/blake2s_utils.rs:179]: Let's create the mask only once at the beginning.

File: src/vm/hints/blake2s_utils.rs [hunk 3]
```diff
@@ -144,6 +152,110 @@ pub fn finalize_blake2s(
...
+    let low = &vm.memory.get_integer(&low_addr)?.clone();
+    let high = &vm.memory.get_integer(&high_addr)?.clone();
+    //Main logic
+    //Declare constant
+    const MASK: u32 = u32::MAX as u32;
+    const B: u32 = 32;
+    //Build first batch of data
+    let mut inner_data = Vec::<BigInt>::new();
+    for i in 0..4 {
+        inner_data.push((high >> (B * (3 - i))) & bigint_u32!(MASK));
```
Comment (Oppen) [inline src/vm/hints/blake2s_utils.rs:231]: Same.

**Review by Oppen — CHANGES_REQUESTED** [review]

File: cairo_programs/blake2s_felts.cairo (hunk 1 above)
[the remark this answers is not in the document]
Comment (fmoletta) [reply cairo_programs/blake2s_felts.cairo:26]: Its arbitrary

File: src/vm/hints/blake2s_utils.rs [hunk 4]
```diff
@@ -41,6 +41,14 @@ fn get_maybe_relocatable_array_from_u32(array: &Vec<u32>) -> Vec<MaybeRelocatabl
...
     new_array
 }
+
+fn get_maybe_relocatable_array_from_bigint(array: Vec<BigInt>) -> Vec<MaybeRelocatable> {
+    let mut new_array = Vec::<MaybeRelocatable>::new();
+    for element in array {
+        new_array.push(MaybeRelocatable::from(element));
+    }
+    new_array
+}
```
[the remark this answers is not in the document]
Comment (fmoletta) [reply src/vm/hints/blake2s_utils.rs]: Fixed!

File: src/vm/hints/blake2s_utils.rs (hunk 2 above)
Comment (fmoletta) [reply src/vm/hints/blake2s_utils.rs]: Fixed!

File: src/vm/hints/blake2s_utils.rs (hunk 3 above)
Comment (fmoletta) [reply src/vm/hints/blake2s_utils.rs]: Fixed!

**Review by Oppen — APPROVED** [review]

**Pull request merged**